test/unit/multipart-test.ts

```diff
@@ -79,9 +108,50 @@ describe(module.id, () => {
 		const writer = bufferWriter();
 		source.transform(multipartParser(heads)).transform(multipartFormatter(heads)).pipe(writer);
 		const result = writer.toBuffer();
-		strictEqual(result.length, 158);
+		strictEqual(result.length, 178);
 		const writer2 = bufferWriter();
 		bufferReader(result).transform(multipartParser(heads)).transform(multipartFormatter(heads)).pipe(writer2);
 		strictEqual(result.toString('binary'), writer2.toBuffer().toString('binary'));
 	});
-});
\ No newline at end of file
+
+    it('basic multipart/form-data', () => {
+        const source = testStreamFormData();
+        const stream = source.transform(multipartParser(headers('form-data')));
+        let part = stream.read();
+        ok(part != null, 'part != null');
+        strictEqual(part.headers.a, 'VA1', 'header A');
+        strictEqual(part.headers.b, 'VB1', 'header B');
+        strictEqual(part.headers['content-type'], 'text/plain', 'content-type');
+        strictEqual(part.headers['content-disposition'], 'form-data; name="c1";', 'content-disposition');
+        let r = part.read();
+        strictEqual(r.toString('binary'), 'C1', 'body C1');
+        r = part.read();
+        strictEqual(r, undefined, 'end of part 1');
+
+        part = stream.read();
+        ok(part != null, 'part != null');
+        strictEqual(part.headers.a, 'VA2', 'header A');
+        strictEqual(part.headers.b, 'VB2', 'header B');
+        strictEqual(part.headers['content-type'], 'text/plain', 'content-type');
+        strictEqual(part.headers['content-disposition'], 'form-data; name="c2";', 'content-disposition');
+        r = part.read();
+        strictEqual(r.toString('binary'), 'C2', 'body C2');
+        r = part.read();
+        strictEqual(r, undefined, 'end of part 2');
+
+        part = stream.read();
+        equal(part, undefined, 'read next part returns undefined');
```

Comment:
```suggestion
        assert.isUndefined(part);
```
Note: I omit the message argument when it paraphrases the test

Reply:
done